api/fastify-app.js

```diff
@@ -0,0 +1,70 @@
+import Fastify from 'fastify'
+import * as Sentry from '@sentry/node'
+import { createRoutes } from './routes/index.js'
+
+export async function createFastifyApp({
+  pgPool,
+  logger,
+  dealIngestionAccessToken,
+  domain
+}) {
+  const fastify = Fastify({
+    logger: {
+      level: 'info',
+      transport: {
+        target: 'pino-pretty'
+      }
+    }
+  })
+
+  // Add shared context to each request
+  fastify.decorateRequest('pgPool', null)
+  fastify.decorateRequest('dealIngestionAccessToken', null)
+  fastify.decorateRequest('domain', null)
```

Comment:
These are constant and don't change from request to request. Therefore, they shouldn't be exposed on the request object.